crates/uv-installer/src/site_packages.rs

```diff
@@ -382,95 +383,28 @@ impl<'a> SitePackages<'a> {
                         }
+                        RequirementSatisfaction::Satisfied => {}
                     }
-
                     // Validate that the installed version satisfies the constraints.
                     for constraint in constraints {
-                        if constraint.name != *distribution.name() {
-                            continue;
-                        }
-
-                        if !constraint.evaluate_markers(self.venv.interpreter().markers(), &[]) {
-                            continue;
-                        }
-
-                        match &constraint.version_or_url {
-                            // Accept any installed version.
-                            None => {}
-
-                            // If the requirement comes from a URL, verify by URL.
-                            Some(pep508_rs::VersionOrUrl::Url(url)) => {
-                                let InstalledDist::Url(installed) = &distribution else {
-                                    return Ok(SatisfiesResult::Unsatisfied(entry.to_string()));
-                                };
-
-                                if installed.editable {
-                                    return Ok(SatisfiesResult::Unsatisfied(entry.to_string()));
-                                }
-
-                                if &installed.url != url.raw() {
-                                    return Ok(SatisfiesResult::Unsatisfied(entry.to_string()));
-                                }
-
-                                // If the requirement came from a local path, check freshness.
-                                if url.scheme() == "file" {
-                                    if let Ok(archive) = url.to_file_path() {
-                                        if !ArchiveTimestamp::up_to_date_with(
-                                            &archive,
-                                            ArchiveTarget::Install(distribution),
-                                        )? {
-                                            return Ok(SatisfiesResult::Unsatisfied(
-                                                entry.to_string(),
-                                            ));
-                                        }
-                                    }
-                                }
-                            }
-
-                            Some(pep508_rs::VersionOrUrl::VersionSpecifier(version_specifier)) => {
-                                // The installed version doesn't satisfy the requirement.
-                                if !version_specifier.contains(distribution.version()) {
-                                    return Ok(SatisfiesResult::Unsatisfied(entry.to_string()));
-                                }
+                        match RequirementSatisfaction::check(
+                            distribution,
+                            constraint.version_or_url.as_ref().map(Into::into),
```

Comment:
I generally prefer `VersionOrUrlRef::from` to make the types clear to the reader. Or does `as_deref` work?

Reply:
`as_deref` doesn't work in this case since it wants to return a `Option<&T>` but we need an `Option<VersionOrUrlRef>`